jira_automation/src/main/java/io/dockstore/jira/MilestoneResolver.java

```diff
@@ -0,0 +1,181 @@
+package io.dockstore.jira;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHMilestone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Attempts to resolve JIRA DOCK ticket fix version mismatches with the corresponding GitHub milestone.
+ * Handles these cases:
+ *
+ * <ol>
+ *     <li>There is a JIRA fix version but no GitHub milestone -- sets the GitHub milestone to the JIRA fix version</li>
+ *     <li>There is a GitHub milestone but no JIRA fix version -- sets the JIRA fix version to the GitHub milestone</li>
+ * </ol>
+ *
+ * <p>It does not handle the case of the JIRA fix version not matching the GitHub milestone. It does
+ * print out a message saying the mismatch needs to be resolved manually. To resolve this automatically,
+ * the program would need to:</p>
+ *
+ * <ol>
+ *     <li>Figure out the timestamp of when the JIRA fix version was last set</li>
+ *     <li>Figure out the timestamp of when the GitHub milestone was set</li>
+ *     <li>Resolve the difference by using the most recently changed.</li>
+ * </ol>
+ *
+ * <p>This is possible, but didn't seem worth the extra work at this point. See
+ * https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issues/#api-rest-api-3-issue-issueidorkey-changelog-get to get
+ * the JIRA change log. Note that it is paginated so the invoker would have to account for that.</p>
+ *
+ */
+public final class MilestoneResolver {
+
+    /**
+     *  Pattern for finding a substring like "Fix Versions: Dockstore 1.15" in GitHub issue body, to extract
+     *  the "1.15"
+     */
+    private static final Pattern FIX_VERSIONS = Pattern.compile("((Fix Versions)|(fixVersions))(:\\s*(Dockstore )?(.*))");
+    private static final int FIX_VERSION_REG_EX_GROUP = 6;
+
+    private static final Logger LOG = LoggerFactory.getLogger(MilestoneResolver.class);
+
+    private MilestoneResolver() { }
+
+    public static void main(String[] args) throws IOException {
+        final List<JiraAndGithub> mismatchedIssues = findMismatchedIssues();
+        if (mismatchedIssues.isEmpty()) {
+            System.out.println("The JIRA fix version and GitHub milestone are in sync for all DOCK issues");
```

Comment:
This class has a `Logger` so could replace the `System.out.println` with `LOG.info` statements, which would remove the need to use `formatted`